scipy/stats/tests/test_contingency.py

```diff
@@ -196,3 +196,62 @@ def test_chi2_contingency_bad_args():
     obs = np.empty((0, 8))
     assert_raises(ValueError, chi2_contingency, obs)
 
+
+def test_bad_association_args():
+    # Invalid Chi Squared Stat Value Type
+    assert_raises(ValueError, association, [[1, 2], [3, 4]], "x", "tschuprow")
+    # Invalid Test Statistic
+    assert_raises(ValueError, association, [[1, 2], [3, 4]], 2.55, "X")
+    # Invalid array shape
+    assert_raises(ValueError, association, [[[1, 2]], [[3, 4]]], 2.55, "cramer")
+
+
+def test_cramersv():
+    # 2d Array
+    obs = np.array([[12, 13, 14, 15, 16],
+                    [17, 16, 18, 19, 11],
+                    [9, 15, 14, 12, 11]])
+    chi2 = chi2_contingency(obs)[0]
+    a = association(obs, chi2_stat=chi2,
+                    stat='v')
+
+    correct = 0.09222412010290792
+    assert_approx_equal(a, correct)
+
+
+def test_tschuprowst():
+    # 2d Array
+    obs = np.array([[12, 13, 14, 15, 16],
```

Comment:
This test and the next ones are almost duplicates of `test_cramersv`. The better way to do this is in a single test:
```
expected = {'v': 0.09222412010290792,
            't': 0.0775509319944633,
            ...}
for method in ('v', 't', 'phi', 'c'):
    obs = np.array([[12, 13, 14, 15, 16],
                    [17, 16, 18, 19, 11],
                    [9, 15, 14, 12, 11]])
    chi2 = chi2_contingency(obs)[0]
    a = association(obs, chi2_stat=chi2,
                    stat=method)
    assert_approx_equal(a, expected[method])
```

Reply:
Completed. Made a separate case for phi which should only be used for 2x2 arrays.